[14.0][ADD] sale_order_invoicing_qty_percentage

pedrobaeza opened the pull request:
**Sales order invoicing by percentage of the quantity**

This module allows to invoice a percentage of the sales order, modifying invoiced quantities according such percentage.

Example: You have a service line to invoice 10 hours. You choose to invoice a 50% of the quantity, so the generated invoice line has 5 hours.

@Tecnativa TT41746

pedrobaeza committed:
[ADD] sale_order_invoicing_qty_percentage

**Sales order invoicing by percentage of the quantity**

This module allows to invoice a percentage of the sales order, modifying
invoiced quantities according such percentage.

Example: You have a service line to invoice 10 hours. You choose to invoice
a 50% of the quantity, so the generated invoice line has 5 hours.

carlosdauden commented on sale_order_invoicing_qty_percentage/models/sale_order.py:
```diff
@@ -0,0 +1,15 @@
+        """If invoicing by quantity percentage, modify quantities."""
+        res = super()._prepare_invoice_line(**optional_values)
+        if self.env.context.get("qty_percentage"):
+            res["quantity"] *= self.env.context["qty_percentage"]
```
Shouldn't we use float_round to avoid potential rounding issues?

pedrobaeza replied:
AFAIK, on create, values are rounded automatically by the precision unit, so no need to do that.

pedrobaeza commented:
/ocabot merge nobump

pedrobaeza commented:
/ocabot merge nobump

pedrobaeza committed:
[FIX] account_invoice_section_sale_order: Avoid problems in integration tests

In OCA repo, now we have this error:

 Traceback (most recent call last):
  File "/__w/account-invoicing/account-invoicing/account_invoice_section_sale_order/tests/test_invoice_group_by_sale_order.py", line 109, in test_create_invoice
    self.assertEqual(line.name, result[line.sequence][0])
KeyError: 1

pedrobaeza commented:
/ocabot merge nobump

pedrobaeza commented:
/ocabot merge nobump
